src/utilities/addAccessors.js

```diff
@@ -29,7 +39,14 @@ const handler = {
  * assert src.b === 'innerChange'
  */
 const addAccessors = (dest, sqZero) => {
-    const ret = [sqZero];
+    if (dest.__isProxy) return dest;
+    let itemZero = sqZero;
+    if (itemZero === undefined) {
+        if (typeof dest !== "object") return dest;
+        if (Array.isArray(dest) && dest.length !== 1) return dest;
+        itemZero = Array.isArray(dest) ? dest[0] : dest;
+    }
+    const ret = [itemZero];
```

Comment:
I guess these changes give `addAccessors` to do a proxy on the `object` too? Do we really need that for objects? I thought the whole point was to be able to use both `seq[0].b` and `seq.b`

Reply:
The point is to make it for sq[0].b and seq.b  The point of the code is to make it easy to create a proxy on various types of objects, without needing to really know what they are - so it works on any of:
{object}
[{object}]
[]
See the unit tests I added.

Reply:
It doesn't add a proxy to the object, it creates a new object that is an array and is a proxy to the passed in object.
That is,
obj = {...attr: value}
addAccessor(obj)
returns a proxy to an array that happens to have access methods for all attributes attr in obj.
That way, the returned object can be accessed either as the array or as the object itself, so that both types work.

Reply:
So does that mean we will end up with all fields as a Proxy? even if they are simple objects?

Reply:
No, only things that are simple objects contained within a sequence need to be proxies.